fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenePostingsTest.java

```diff
@@ -0,0 +1,414 @@
+            recordStore.saveRecord(createSimpleDocument(1624L, "Document 2", 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, "NonDocument 3", 2));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.deleteRecord(Tuple.from(1623L));
+            recordStore.deleteRecord(Tuple.from(1624L));
+            recordStore.deleteRecord(Tuple.from(1547L));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext(contextProps)) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            final RecordQuery query = buildQuery("Document", Collections.emptyList(), SIMPLE_DOC);
+            queryAndAssertFields(query, "text", Map.of());
+            LuceneIndexTestUtils.mergeSegments(recordStore, index);
+            try (FDBDirectory directory = new FDBDirectory(recordStore.indexSubspace(index), context, index.getOptions())) {
+                // When deleting all docs from the index, the first segment (_0) was merged away and the last segment (_1) gets removed
+                assertFieldCountPerSegment(directory, List.of("_0", "_1"), List.of(0, 0));
+            }
+            assertTrue(timer.getCounter(LuceneEvents.Counts.LUCENE_DELETE_POSTINGS).getCount() > 0);
+        }
+    }
+
+    @Test
+    void testComplexDocManyFields() throws Exception {
+        // Use a complex index with several fields
+        Index index = INDEXED_COMPLEX_WITH_OPT_POSTINGS;
+
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            recordStore.saveRecord(LuceneIndexTestUtils.createComplexDocument(1623L, "Hello", "Hello 2", 5, 12, false, 7.123));
+            recordStore.saveRecord(LuceneIndexTestUtils.createComplexDocument(1624L, "Hello record", "Hello record 2", 6, 13, false, 8.123));
+            recordStore.saveRecord(LuceneIndexTestUtils.createComplexDocument(1625L, "Hello record layer", "Hello record layer 2", 7, 14, true, 9.123));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            RecordQuery query = buildQuery("record", Collections.emptyList(), COMPLEX_DOC);
+            // This doc has the PK as (Group, docID)
+            queryAndAssertFieldsTuple(query, "text", Map.of(
+                    Tuple.from(6, 1624L), "Hello record",
+                    Tuple.from(7, 1625L), "Hello record layer"));
+            // query on the other fields
+            query = buildQuery("text2:layer", Collections.emptyList(), COMPLEX_DOC);
+            queryAndAssertFieldsTuple(query, "text2", Map.of(
+                    Tuple.from(7, 1625L), "Hello record layer 2"));
+            query = buildQuery("group:5", Collections.emptyList(), COMPLEX_DOC);
```

Comment:
This query and the one below seem irrelevant to the postings format, and could be removed.